frontend/src/component/project/Project/CreateProject/NewProjectForm.tsx

```diff
@@ -75,15 +72,66 @@ const FormActions = styled(StyledFormSection)(({ theme }) => ({
 
 const CREATE_PROJECT_BTN = 'CREATE_PROJECT_BTN';
 
-export const NewProjectForm = () => {
-    const navigate = useNavigate();
+type FormProps = {
+    projectId: string;
+    projectName: string;
+    projectDesc: string;
+    projectStickiness?: string;
+    featureLimit?: string;
+    featureCount?: number;
+    projectMode?: string;
+    setProjectStickiness?: React.Dispatch<React.SetStateAction<string>>;
+    setProjectId: React.Dispatch<React.SetStateAction<string>>;
+    setProjectName: React.Dispatch<React.SetStateAction<string>>;
+    setProjectDesc: React.Dispatch<React.SetStateAction<string>>;
+    setFeatureLimit?: React.Dispatch<React.SetStateAction<string>>;
+    setProjectMode?: React.Dispatch<React.SetStateAction<ProjectMode>>;
+    handleSubmit: (e: any) => void;
+    errors: { [key: string]: string };
+    mode: 'Create' | 'Edit';
+    clearErrors: () => void;
+    validateProjectId: () => void;
+};
```

Comment:
These are all the props from the old form. I've put them here as a template. We'll use the ones we need and remove the ones we don't as we go.